portal-frontend/src/app/features/edit-submission/parcel-details/parcel-entry/parcel-entry.component.ts

```diff
@@ -172,28 +180,64 @@ export class ParcelEntryComponent implements OnInit {
   }
 
   onChangeParcelType($event: MatButtonToggleChange) {
-    if ($event.value === 'CRWN') {
-      this.searchBy.setValue(null);
-      this.pidPinPlaceholder = '';
-      this.isCrownLand = true;
-      this.pid.setValidators([]);
-      this.purchaseDate.reset();
-      this.purchaseDate.disable();
+    const dirtyForm =
+      this.legalDescription.value ||
+      this.mapArea.value ||
+      this.pid.value ||
+      this.pin.value ||
+      this.purchaseDate.value ||
+      this.isFarm.value ||
+      this.civicAddress.value;
+
+    const changeParcelType = () => {
+      if ($event.value === this.PARCEL_OWNERSHIP_TYPES.CROWN) {
+        this.searchBy.setValue(null);
+        this.pidPinPlaceholder = '';
+        this.isCrownLand = true;
+        this.pid.setValidators([]);
+        this.purchaseDate.disable();
+      } else {
+        this.searchBy.setValue('pid');
+        this.pidPinPlaceholder = 'Type 9 digit PID';
+        this.isCrownLand = false;
+        this.pid.setValidators([Validators.required]);
+        this.crownLandOwnerType.setValue(null);
+        this.purchaseDate.enable();
+      }
+
+      this.updateParcelOwners([]);
+      this.filteredOwners = this.mapOwners(this.owners);
+      this.pid.updateValueAndValidity();
+    };
+
+    if (dirtyForm && this.isCrownLand !== null) {
+      this.dialog
+        .open(ParcelEntryConfirmationDialogComponent, {
+          panelClass: 'no-padding',
+          disableClose: true,
+        })
+        .beforeClosed()
+        .subscribe(async (result) => {
+          if (result) {
+            this.onReset();
+            return changeParcelType();
+          } else {
+            const newParcelType = this.parcelType.getRawValue();
+
+            const prevParcelType =
+              newParcelType === this.PARCEL_OWNERSHIP_TYPES.CROWN
+                ? this.PARCEL_OWNERSHIP_TYPES.FEE_SIMPLE
+                : this.PARCEL_OWNERSHIP_TYPES.CROWN;
+
+            this.parcelType.setValue(prevParcelType);
+          }
+        });
     } else {
-      this.searchBy.setValue('pid');
-      this.pidPinPlaceholder = 'Type 9 digit PID';
-      this.isCrownLand = false;
-      this.pid.setValidators([Validators.required]);
-      this.crownLandOwnerType.setValue(null);
-      this.purchaseDate.enable();
+      return changeParcelType();
     }
-
-    this.updateParcelOwners([]);
-    this.filteredOwners = this.mapOwners(this.owners);
-    this.pid.updateValueAndValidity();
   }
 
-  async attachFile(file: FileHandle, documentType: DOCUMENT_TYPE, parcelUuid: string) {
+  async attachFile(file: FileHandle, _documentType: DOCUMENT_TYPE, parcelUuid: string) {
```

Comment:
If this parameter isnt used, should it be deleted?

Reply:
Looks like we can remove it. The document type is specified in the service, we probably just forgot to update here. I'll delete it here!